driver-core/src/main/com/mongodb/internal/operation/retry/AttachmentKeys.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.internal.operation.retry;
+
+import com.mongodb.annotations.Immutable;
+import com.mongodb.bulk.BulkWriteResult;
+import com.mongodb.internal.async.function.LoopState.AttachmentKey;
+import com.mongodb.internal.operation.MixedBulkWriteOperation.BulkWriteTracker;
+import org.bson.BsonDocument;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.mongodb.assertions.Assertions.assertTrue;
+import static com.mongodb.assertions.Assertions.fail;
+
+/**
+ * @see AttachmentKey
```

Comment:
I am not sure what requires documenting here. The `AttachmentKeys` class with its static methods that return `AttachmentKey` values should have been an `enum` if it were not for Java forbidding to have instances of the same `enum` with different values of a type parameter. So I would like to write
```java
public enum AttachmentKeys<V> implements AttachmentKey<V> {
    OPERATION_NAME<String>,
    MAX_WIRE_VERSION<Integer>,
    ...
}
```
but it is not possible in Java. However, if it were possible, would you actually need some sort of documentation here, given that this enum is an internal class, and the ways keys are used in the code is completely up to the code? If yes, could you please tell what do you think should have been documented?

P.S. What's even worse, no `enum` can be considered a [value-based](https://docs.oracle.com/javase/8/docs/api/java/lang/doc-files/ValueBased.html) class because [_"there is only one instance of each enum constant, it is permitted to use the == operator in place of the equals method"_](https://docs.oracle.com/javase/specs/jls/se17/html/jls-8.html#jls-8.9.1). This is a very unfortunate situation (╯°□°）╯︵ ┻━┻, given that the whole idea of a `enum` in Java is to express a set of values.

Reply:
Added a documentation comment to the `AttachmentKeys` class.